modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -265,3 +269,28 @@ func (k Keeper) DeleteMiddlewareEnabled(ctx sdk.Context, portID, connectionID st
 	store := ctx.KVStore(k.storeKey)
 	store.Delete(icatypes.KeyIsMiddlewareEnabled(portID, connectionID))
 }
+
+// GetAuthority returns the ica/controller submodule's authority.
+func (k Keeper) GetAuthority() string {
+	return k.authority
+}
+
+// GetParams returns the current ica/controller submodule parameters.
+func (k Keeper) GetParams(ctx sdk.Context) types.Params {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get([]byte(types.ParamsKey))
+	if bz == nil {
+		panic("ica/controller params are not set in store")
+	}
```

Comment:
I would still advocate for the following:

```suggestion
	if len(bz) == 0 {
		return types.Params{}
	}
```

This came up in the client params PR also - see https://github.com/cosmos/ibc-go/pull/3640#discussion_r1204210620

Reply:
But this would be inconsistent with the host and transfer PRs. Should we at least preserve consistency within the ica module? XD